### PR TITLE
Revise Handle Limited Open Connections

### PR DESCRIPTION
**Pull request by sgothel**
* Resolves: #9833  refining PR #9916
* Target version: master 

### Summary
This track follows up on PR #10334 and may include PR for issue #10337

#### Functional
- Fixes all review comments in: https://github.com/CollaboraOnline/online/pull/10334 particularly
- Removes WS slow ping connection close feature
  - The added implementation as well as the robust alternative #10337 has been rejected
- Use system maximum TCP connections to limit new connections on `SocketPoll`
  - Drop relation to maximum sessions
  - Use Linux system values min(tcp_max_orphans, net.netfilter.nf_conntrack_max)
    - [tcp_max_orphans](https://www.kernel.org/doc/html/latest/networking/ip-sysctl.html)
    - [net.netfilter.nf_conntrack_max](https://www.kernel.org/doc/html/latest/networking/nf_conntrack-sysctl.html)

#### Style
- Removes immutable member variables on frequently created objects (copies of global settings)
  - In particular `net::Defaults` related
- Fixes (some) naming conventions breakage
- Removes redundant boolean `SocketPoll::_limitedConnections`, use query method instead
- Removes `SocketPoll`'s `itemsErased` accounting while erasing
- Removes redundant `Socket::isClosed()`

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: common/FileUtil.hpp [hunk 1]
```diff
@@ -183,6 +184,39 @@
...
+        if (dataLen <= 1) // no space for data + EOS
+            return defaultValue;
+        errno = 0;  // Flush previous error indicator. Reminder: errno is thread-local
+        const int fd = ::open(path.c_str(), O_RDONLY);
+        if (fd < 0)
+            return defaultValue;
+
+        ::memset(data, 0, dataLen);
+        ssize_t consumed = 0;
+        while ( dataLen-consumed-1 /* ex-EOS */ > 0 )
```
[the remark this answers is not in the document]
Comment (sgothel) [reply common/FileUtil.hpp:202]: invalid due to pre-validated dataLen.

**Comment by sgothel** [comment]
All checks have passed .. will rebase to master now .. (no merge yet, duh!)

File: net/Socket.cpp [hunk 2]
```diff
@@ -726,28 +726,14 @@ int SocketPoll::poll(int64_t timeoutMaxMicroS)
 
         if (itemsErased)
         {
-            const size_t itemsErasedPre = itemsErased;
-            itemsErased = 0; // correcting itemsErased
+            LOG_TRC("Scanning to removing " << itemsErased << " defunct sockets from "
+                    << _pollSockets.size() << " sockets");
```
Comment (mmeeks) [inline net/Socket.cpp:728]: I think what I am looking for here is a revert of the complexity inherent in:

80246f7ac1140257d14162504d3550909d6ba681

Reverting that patch as step #1 in its entirety - and then adding whatever it is we wanted to do back again would make me significantly happier. Lets get the revert in - and then re-think that from scratch - I am suspicious of all the changes to the arithemetic there - the itemsAdded size comparisons etc. etc. if there is a big-fix or benefit from each change to the logic there - it should be split out, reviewed and ideally unit tested separately. We should bin the lock, atomic, etc. etc. there too.

**Comment by sgothel** [comment]
A weirdo merge conflict  occurred via github here, which wasn't existing locally (rebased to master). 
"Fun" ..

**Comment by mmeeks** [comment]
So - there is once again a huge amount of change here - a ton of it is not related to reverting back to the state before we started making huge amounts  of change. Again - I want to restore us to a known-safe state tested on millions of users - and then we take very, very small steps, incrementally, without huge and questionable cosmetic changes on top to improve the code.

Perhaps it is nice to do some stream logging operator improvement - perhaps that makes people feel good - but to do that mixed with functional change, and/or to do it at all now - when the priority is getting us back to a known good state is highly unhelpful.

Perhaps it would be better to have a more simple approach. Revert everything that went in recently - and then go over it again - breaking it down properly, thinking through the design carefully, and so on. At this point I'm basically not interested in any new features - connection limits, trottling, closing websockets, etc. - I want us back where we were.

File: net/Socket.cpp (hunk 2 above)
Comment (sgothel) [reply net/Socket.cpp:728]: it was just logging I added. Not required.

**Comment by sgothel** [comment]
rebased again, mmeeks branch (w/ same commits) got merged, hence earlier confusion (collision).

**Comment by sgothel** [comment]
Will be revised to fully revert 2 commits which have been partially reverted and reapply changes on top of vanilla.

**Pull request closed without merging**